src/contracts/core/AllocationManager.sol

```diff
@@ -313,101 +285,49 @@ contract AllocationManager is
+                require(freeInfo.freeMagnitude >= magnitudeToAllocate, InsufficientAllocatableMagnitude());
+                freeInfo.freeMagnitude -= magnitudeToAllocate;
 
-                // 2. allocate magnitude which will take effect in the future 21 days from now
-                magnitudeUpdates.push({key: allocationEffectTimestamp, value: allocation.magnitudes[i]});
+                // 2. Update the effectTimestamp for the allocation
+                effectTimestamp = allocationEffectTimestamp;           
             }
-        }
-    }
 
-    /**
-     * @notice Get the number of queued allocation/deallocations for the given (operator, strategy, operatorSetKey) tuple
-     * @param operator address to get queued allocation/deallocations  for
-     * @param strategy the strategy to get queued allocation/deallocations  for
-     * @param operatorSetKey the encoded operatorSet to get queued allocation/deallocations for
-     * @dev Does not make an assumption on the number of pending allocations or deallocations at a time
-     */
-    function _getPendingAllocationsAndDeallocations(
-        address operator,
-        IStrategy strategy,
-        bytes32 operatorSetKey
-    ) internal view returns (uint256 numPendingAllocations, uint256 numPendingDeallocations) {
-        // Get pending allocations
-        Snapshots.History storage magnitudeUpdates = _magnitudeUpdate[operator][strategy][operatorSetKey];
-
-        // Read current magnitude's respective array index and length.
-        (,uint256 latestAllocPos, uint256 allocLength) =
-            magnitudeUpdates.upperLookupRecentWithPos(uint32(block.timestamp));
-
-        if (latestAllocPos < allocLength - 1) {
-            numPendingAllocations = allocLength - 1 - latestAllocPos;
-        }
-        // Else, latestAllocPos >= allocLength - 1;
-        // latestAllocPos cannot be greater than length - 1, thus when pos == length - 1 there are no pending allocations
-
-        // Get pending deallocations
-        uint256 deallocationsLength = _queuedDeallocationIndices[operator][strategy][operatorSetKey].length;
-        for (uint256 i = deallocationsLength; i > 0; --i) {
-            // index of pendingFreeMagnitude/deallocation to check for slashing
-            uint256 index = _queuedDeallocationIndices[operator][strategy][operatorSetKey][i - 1];
-
-            // If completableTimestamp is greater than completeUntilTimestamp, break
-            if (block.timestamp < _pendingFreeMagnitude[operator][strategy][index].completableTimestamp) {
-                ++numPendingDeallocations;
-            } else {
-                break;
-            }
+            // Allocate magnitude which will take effect at the `effectTimestamp`
+            
+            _operatorMagnitudeInfo[operator][allocation.strategy][operatorSetKey] = MagnitudeInfo({
+                currentMagnitude: currentMagnitude,
+                pendingMagnitudeDiff: int64(allocation.magnitudes[i]) - int64(currentMagnitude),
```

Comment:
I feel like all math should be using 256 bits, then we safecast to the smaller sizes when mutating.